File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):

    answer = []
    time = 0
    count = 0
    
    while len(progresses)> 0:
        if (progresses[0] + time*speeds[0]) >= 100: #각 기능 별 진행정도 + 시간*각 기능 별 속도가 100 이상이라면
            progresses.pop(0) #첫번째 꺼내기(pop에서 꺼낸 원소는 progresses에서 제거됨)
            speeds.pop(0) #첫번째 꺼내기(pop에서 꺼낸 원소는 speeds에서 제거됨)
            count += 1 #카운트
            
        else: #각 기능 별 진행정도 + 시간*각 기능 별 속도가 100 이하라면
            if count > 0: #count가 0 초과라면
                answer.append(count) #answer에 count 추가
                count = 0 #예시 1번의 경우 7일째에 완료되는 기능은 2개이기 때문에 count 초기화
            time += 1 #if조건에 맞게 하기 위해 루프가 총 7번 돌고 time이 7되면 첫 번째 if 조건문 실행
    answer.append(count)
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;


vector<int>solution(vector<int> progresses, vector<int> speeds)
{
    vector<int>answer;
    vector<int>day;
    int max_day = 0;


    for (int i = 0; i < progresses.size(); i++) {
        day.push_back(ceil((float)(100 - progresses[i]) / speeds[i])); //진행 완료까지 걸리는 일 수 계산
        //float으로 캐스팅하는게 매우 중요!!!!!!!
       
        if (day[i] > max_day) { //day[i]가 max_day보다 크면
            answer.push_back(1); //answer의 뒤에 1 추가
            max_day = day[i];//day[i]를 max_day로 선언
        }
        else { //아닌 경우 = day[i]가 max_day와 같거나 작은 경우 혹은 answer가 공백인 경우
            answer.back()++;//answer의 뒤의 요소에 1 증가(큐의 원리 선입선출)
        }
    }

    return answer;
}
